### PR TITLE
Add Precision Landing tests

### PR DESCRIPTION
**Pull request by khancyr**
need https://github.com/ArduPilot/ardupilot/pull/9199

rebase and cleanup from https://github.com/ArduPilot/ardupilot/pull/8932

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -267,10 +267,12 @@ def fly_square(self, side=50, timeout=300):
         self.save_wp()
 
         # switch back to stabilize mode
-        self.set_rc(3, 1500)
         self.mavproxy.send('switch 6\n')
         self.wait_mode('STABILIZE')
 
+        # increase throttle because we're about to pitch:
+        self.set_rc(3, 1550)
```
Comment (khancyr) [inline Tools/autotest/arducopter.py]: @peterbarker Do you remember why you raise throttle ? we are in Loiter, so the copter is just climbing infinitly !

**Comment by peterbarker** [comment]
On Tue, 28 Aug 2018, Pierre Kancir wrote:

> >          self.mavproxy.send('switch 6\n')
>          self.wait_mode('STABILIZE')
> 
> +        # increase throttle because we're about to pitch:
> +        self.set_rc(3, 1550)
> 
> @peterbarker Do you remember why you raise throttle ? we are in Loiter, so
> the copter is just climbing infinitly !

The comment indicates why - we're in stabilize mode here and we were 
running into the ground when we pitched.

The rc should be reset to 1500 when we stop pitching...

**Comment by khancyr** [comment]
Right ... I miss that we were in stabilize ... I reduce a little the throttle

**Comment by rmackay9** [comment]
I think I'll leave this to @peterbarker to merge..

**Comment by khancyr** [comment]
rebased

**Comment by khancyr** [comment]
split with https://github.com/ArduPilot/ardupilot/pull/9465

**Comment by amilcarlucas** [comment]
Needs a rebase, otherwise anything holding this up @peterbarker ?

**Comment by peterbarker** [comment]
@amilcarlucas This was actually merged somehow else :-)

**Pull request closed without merging**